src/ToolInfo.php

```diff
@@ -73,31 +72,31 @@ public static function isInstalledByComposer()
         static $result;
 
         if (null === $result) {
-            $result = !self::isInstalledAsPhar() && file_exists(self::getScriptDir().self::COMPOSER_INSTALLED_FILE);
+            $result = !self::isInstalledAsPhar() && file_exists(self::getComposerInstalledFile());
         }
 
         return $result;
     }
 
-    private static function getScriptDir()
+    private static function getComposerInstalledFile()
     {
         static $result;
 
         if (null === $result) {
-            $script = $_SERVER['SCRIPT_NAME'];
-
-            if (is_link($script)) {
-                $linkTarget = readlink($script);
-
-                // If the link target is relative to the link
-                if (false === realpath($linkTarget)) {
-                    $linkTarget = dirname($script).'/'.$linkTarget;
+            $composerInstalledFile = '/composer/installed.json';
+            $checkedPaths = array(
+                // this project itself at ./src
+                __DIR__.'/../vendor'.$composerInstalledFile,
+                // the project installed through composer at ./vendor/friendsofphp/php-cs-fixer/src
+                __DIR__.'/../../..'.$composerInstalledFile,
+            );
+
+            foreach ($checkedPaths as $path) {
+                $result = $path;
```

Comment:
This should be done in the `if`, otherwise the last checked file will be returned even if it doesn't exist.